src/Razor/src/Microsoft.NET.Sdk.Razor/GenerateAspNetCoreStaticAssetsManifest.cs

```diff
@@ -0,0 +1,179 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Xml;
+using System.Xml.Linq;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace Microsoft.AspNetCore.Razor.Tasks
+{
+    public class GenerateAspNetCoreStaticAssetsManifest : Task
+    {
+        private const string ContentRoot = nameof(ContentRoot);
+        private const string BasePath = nameof(BasePath);
+        
+        [Required]
+        public string TargetManifestPath { get; set; }
+
+        [Required]
+        public ITaskItem[] ContentRootDefinitions { get; set; }
+
+        public override bool Execute()
+        {
+            if(!ValidateArguments())
+            {
+                return false;
+            }
+
+            return ExecuteCore();
+        }
+
+        private bool ExecuteCore()
+        {
+            var document = new XDocument(new XDeclaration("1.0", "utf-8", "yes"));
+            var root = new XElement(
+                "AspNetCoreStaticAssets",
+                new XAttribute("Version", "1.0"),
+                CreateNodes());
+
+            document.Add(root);
+
+            var settings = new XmlWriterSettings
+            {
+                Encoding = Encoding.UTF8,
+                CloseOutput = true,
+                OmitXmlDeclaration = true,
+                Indent = true,
+                NewLineOnAttributes = false,
+                Async = true
+            };
+
+            using (var xmlWriter = GetXmlWritter(settings))
+            {
+                document.WriteTo(xmlWriter);
+            }
+
+            return true;
+        }
+
+        private IEnumerable<XElement> CreateNodes()
+        {
+            var nodes = new List<XElement>();
+            for (var i = 0; i < ContentRootDefinitions.Length; i++)
+            {
+                var contentRootDefinition = ContentRootDefinitions[i];
+                var basePath = contentRootDefinition.GetMetadata(BasePath);
+                var contentRoot = contentRootDefinition.GetMetadata(ContentRoot);
+                
+                // At this point we already know that there are no elements with different base paths and same content roots
+                // or viceversa. Here we simply skip additional items that have the same base path and same content root.
+                if (!nodes.Exists(e => e.Attribute(BasePath).Value.Equals(basePath, StringComparison.OrdinalIgnoreCase)))
+                {
+                    nodes.Add(new XElement("ContentRoot",
+                        new XAttribute("BasePath", basePath),
+                        new XAttribute("Path", contentRoot)));
```

Comment:
What's the difference here between a `BasePath` and a `Path`?

Reply:
Path is the physical path to the content root on disk.
Base path is the prefix under which the content needs to be exposed.
For example, when you create a wwwroot folder on a project, we don’t want to expose the content directly under wwwroot on the consuming app, but under something like _content/library name.
BasePath is that prefix.

Reply:
For example, if you have a library Awesomesauce and you reference it (either like a project or a package) we want to expose the assets by default under "_content/Awesomesauce".

So wwwroot\index.js on Awesomesauce gets exposed as _content/Awesomesauce/index.js when consumed either as a package reference or as a project reference.

Reply:
In the default case, the built-in targets will set `BasePath` to `_content/<packagename>`, right?

Reply:
@SteveSandersonMS Yes, but I want to "sanitize" it a bit by removing anything that is not alphanumeric and lowercasing the package name.

I will do that as part of the next PR wave